src/kits/serial/SerialPortEmu.sedona

```diff
@@ -0,0 +1,85 @@
+//
+// Copyright (c) 2016 Andrey Skvortsov <andrej.skvortzov@gmail.com
+//
+// Licensed under the Academic Free License version 3.0
+//
+
+class SerialPortEmu extends SerialPortNative
```

Comment:
If we make both SerialPortEmu and SerialPort to inherit from SerialPortNative class and be final, then the logics of SerialPortEmu will be simplified, these **if** statements to check if isVirtualPort will be unnecessary.

Reply:
I think, this will not work. It was intended not to have two separate serial port classes: real and emulated, but have only one class SerialPort and use emulated port for debugging without changing any code.  Look at test/SerialTest.sedona.

Reply:
Ok, I see.